src/ip3country.js

```diff
@@ -13,13 +15,27 @@ class CountryLookup {
    242.n2.n3.c: if n >= 240 but < 65536. n2 being lower order byte
    243.n2.n3.n4.c: if n >= 65536. n2 being lower order byte
   */   
-  async init() {
+  init: async function() {
+    if (this._initialized) {
+      console.warn("ip3country already initialized");
+      return;
+    }
+
     const buffer = await fs.readFile(path.resolve(__dirname, '../bin/ip_supalite.bin'));
-    
-    this.countryCodes = [];
-    this.ipRanges = [];
-    this.countryTable = [];
+    this.initWithBuffer(buffer);
+  },
```

Comment:
If you call init again before the file finishes reading, initWithBuffer will get called multiple times. We solve this for some things with an "InitializationPromise" in addition to the initialized flag for active init that we set while initializing, and await if it exists

Reply:
Hmm - what I really wanted to prevent was reentrancy.  Let me take care of this.